### PR TITLE
Map set up

### PR DESCRIPTION
**Pull request by mluena**
### Overview

This pull request enhances the map functionality by introducing the display of layers. The changes include:

- Creation of a new service and three hooks:
     1. To fetch data for all available layers.
     2. To fetch data for a specific layer based on its ID.
     3.  To fetch monitors.
- Current temporary approach: Frontend handles filtering in layers due to API limitations; this should be revisited once the API supports filtering.
- Populated the legend with basic data about monitors, geo-stories, and layers.
- Updates the URL when a layer is selected.
- Please note that currently, layers cannot be removed from the map; they can only be replaced.
- No tests are included in this pull request as the workflow is not yet complete.
- This PR also includes a bit of cleaning on non-used components (sorry)

** There are a couple of endpoints created in the next API that  don't want to remove in case real API gets down at some point

### Testing instructions

_Go to the map section by clicking on the header, and add layers to the map. Once you have selected one layer you can remove it just replace it (limitations due to the early stage of the app)_

### Feature relevant tickets

_[Link to the related task manager tickets](https://vizzuality.atlassian.net/browse/OEMC-23?atlOrigin=eyJpIjoiN2M4OWEzMGQ3YjYyNGQ0OGIzZWJiZjNiYzA5MjNhYjQiLCJwIjoiaiJ9)_

---

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] Update CHANGELOG
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist

### CODE REVIEW

**Comment by davidsingal** [comment]
Screenshot from 2023-09-06 08-44-36
Did you check the layout on large screens?

File: src/hooks/types.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+export type UseParamsOptions = Readonly<{
```
Comment (davidsingal) [inline src/hooks/types.tsx:1]: types should be in a folder called `types` inside `src`. Something like `types/entities.d.ts` for example.

**Review by davidsingal — CHANGES_REQUESTED** [review]

File: src/types/datasets.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
+export type UseParamsOptions = Readonly<{
```
Comment (davidsingal) [inline src/types/datasets.tsx:1]: preferably types files should have the extension `d.ts`.

File: src/components/map/legend/mock.tsx [hunk 3]
```diff
@@ -1,14 +1,4 @@
-import Icon from 'components/icon';
```
Comment (davidsingal) [inline src/components/map/legend/mock.tsx:1]: Only files with JSX syntax have the `.tsx` extension. If there is no component (as in this file), it should be `.ts`.

File: components.json [hunk 4]
```diff
@@ -10,6 +10,6 @@
   },
   "aliases": {
     "components": "@/components",
-    "utils": "@/lib/utils"
+    "utils": "@/lib/classnames"
```
[the remark this answers is not in the document]
Comment (davidsingal) [reply components.json:13]: Why do you prefer to keep this?

File: src/app/layout.tsx [hunk 5]
```diff
@@ -1,13 +1,21 @@
...
 import '@/styles/globals.css';
-export const metadata: Metadata = {
-  title: 'Hub - Open-Earth-Monitor Cyberinfrastructure',
-  description: '...',
-};
+
+// export const metadata: Metadata = {
+//   title: 'Hub - Open-Earth-Monitor Cyberinfrastructure',
+//   description: '...',
+// };
```
Comment (davidsingal) [inline src/app/layout.tsx:18]: Try to not keep comments in the final version of the PR

File: src/constants/api-payloads/layers.ts [hunk 6]
```diff
@@ -0,0 +1,144 @@
+export const LAYERS = [
```
Comment (davidsingal) [inline src/constants/api-payloads/layers.ts:1]: for mocking purposes my recommendation is working with JSON files with the format provided from the API

File: src/services/api/index.ts [hunk 7]
```diff
@@ -1,8 +1,9 @@
 import axios from 'axios';
 
 const API = axios.create({
-  baseURL: `${process.env.NEXT_PUBLIC_API_URL}`,
-  headers: { 'Content-Type': 'application/json' },
+  baseURL: process.env.NEXT_PUBLIC_API_URL,
+  maxBodyLength: Infinity,
```
Comment (davidsingal) [inline src/services/api/index.ts:5]: option no needed after once the CORS issues has been fixed

**Review by davidsingal — CHANGES_REQUESTED** [review]

**Comment by davidsingal** [comment]
Additionally, the layout still has scrolls on large screens.
Screenshot 2023-09-08 at 10 52 28

File: src/app/layout.tsx (hunk 5 above)
Comment (mluena) [reply src/app/layout.tsx:18]: I am not sure if it's the final

**Pull request merged**